src/ui/RadarMarker.ts

```diff
@@ -121,6 +121,20 @@ class RadarMarker extends maplibregl.Marker {
     }
     return super.remove();
   }
+
+  _onMapClick(e: maplibregl.MapMouseEvent): void {
+    const targetElement = e.originalEvent.target;
+    const element = this._element;
+
+    if (targetElement === element || element.contains(targetElement as any)) {
+      // TODO: Need to stop propagation somehow
+      // can use addEventListener in the constructor instead but then the togglePopup logic will be kinda wonky
+      if (this._popup) {
+        this.togglePopup();
+      }
+      this.fire('click', this);
+    }
+  }
```

Comment:
@kochis i screwed up the commits and the [previous PR](https://github.com/radarlabs/radar-sdk-js/pull/154) closed but to address your comment this is a method override

Reply:
this is the maplibre gl js implementation
https://github.com/maplibre/maplibre-gl-js/blob/3000ba109488c5503ce7719066ae0bd8a8100960/src/ui/marker.ts#L472-L479

Reply:
TODO: do not do method override. Add a click handler to the marker html element